cpp/include/cudf/hashing.hpp

```diff
@@ -42,7 +42,12 @@ enum class hash_id {
   HASH_IDENTITY = 0,   ///< Identity hash function that simply returns the key to be hashed
   HASH_MURMUR3,        ///< Murmur3 hash function
   HASH_SPARK_MURMUR3,  ///< Spark Murmur3 hash function
-  HASH_MD5             ///< MD5 hash function
+  HASH_MD5,            ///< MD5 hash function
+  HASH_SHA1,           ///< SHA-1 hash function
+  HASH_SHA224,         ///< SHA-224 hash function
+  HASH_SHA256,         ///< SHA-256 hash function
+  HASH_SHA384,         ///< SHA-384 hash function
+  HASH_SHA512          ///< SHA-512 hash function
```

Comment:
This enum was meant to be deprecated (or at least moved) so I don't think we should be adding new values.
We intend to deprecate the enum-based hash function here: https://github.com/rapidsai/cudf/pull/14391/files#diff-bec3f1500ce3eafc694b02cc2c575ac8b4edb66b356b7e9413601e2c183b06f8L66 in a separate PR.

Reply:
I pushed 0bf01fc for this, but I'll need to do a bit more cleanup (and haven't had time to build those changes locally yet).